### PR TITLE
flattening: Release operator on new tag

### PR DESCRIPTION
**Pull request by fiunchinho**
Towards https://github.com/giantswarm/giantswarm/issues/7819

### CODE REVIEW

**Comment by nprokopic** [comment]
Do we maybe want to move to `architect-orb` straight away? See https://github.com/giantswarm/aws-operator/pull/2143.

**Comment by fiunchinho** [comment]
> Do we maybe want to move to `architect-orb` straight away? See [giantswarm/aws-operator#2143](https://github.com/giantswarm/aws-operator/pull/2143).

I think [we can't just yet](https://github.com/giantswarm/aws-operator/pull/2143#issuecomment-590235782).

File: .circleci/config.yml [hunk 1]
```diff
@@ -20,31 +20,45 @@ e2eTest: &e2eTest
...
+          date +"%Y" > /tmp/year
+          date +"%m" > /tmp/month
+          date +"%d" > /tmp/day
+      - restore_cache:
+          keys:
+            - go-cache-v1-{{ checksum "/tmp/year" }}-{{ checksum "/tmp/month" }}-{{ checksum "/tmp/day" }}
+            - go-cache-v1-{{ checksum "/tmp/year" }}-{{ checksum "/tmp/month" }}
+            - go-cache-v1-{{ checksum "/tmp/year" }}
+
+      - run: ./architect build
```
Comment (nprokopic) [inline .circleci/config.yml]: We should have step names, see https://github.com/giantswarm/fmt/blob/master/circleci/all-circleci-best-practices.md.
For example:
```suggestion
      - run:
          name: Build using architect
          command: ./architect build
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -20,31 +20,45 @@ e2eTest: &e2eTest
...
   build:
     machine: true
     steps:
-    - checkout
-    - run:
-        name: Install architect
-        command: |
+      - checkout
+
+      - run: |
```
Comment (nprokopic) [inline .circleci/config.yml]: Build step name missing, e.g.:
```suggestion
      - run:
          name: Install architect
          command: |
            wget -q $(curl -sS -H "Authorization: token $RELEASE_TOKEN" https://api.github.com/repos/giantswarm/architect/releases/latest | grep browser_download_url | head -n 1 | cut -d '"' -f 4)
            chmod +x ./architect
            ./architect version
```

**Pull request merged**